src/algorithms.jl

```diff
@@ -207,31 +213,20 @@ function MSC(h)
     un = 0.19783982482140777 #4.0X/(X+15.0Y+3.0Z)
     vn = 0.46833630293240970 #9.0Y/(X+15.0Y+3.0Z)
 
-    #sRGB matrix
-    M = [0.4124564  0.3575761  0.1804375;
-         0.2126729  0.7151522  0.0721750;
-         0.0193339  0.1191920  0.9503041]'
-
-    m_tx=M[t,1]
-    m_ty=M[t,2]
-    m_tz=M[t,3]
-    m_px=M[p,1]
-    m_py=M[p,2]
-    m_pz=M[p,3]
+    m_tx, m_ty, m_tz = M_RGB2XYZ[:, t]
+    m_px, m_py, m_pz = M_RGB2XYZ[:, p]
 
     f1 = 4alpha*m_px+9beta*m_py
     a1 = 4alpha*m_tx+9beta*m_ty
     f2 = m_px+15m_py+3m_pz
     a2 = m_tx+15m_ty+3m_tz
 
     cp=((alpha*un+beta*vn)*a2-a1)/(f1-(alpha*un+beta*vn)*f2)
+    cpc = clamp01(srgb_compand(cp))
 
-    col = zeros(3)
-    col[p] = clamp01(srgb_compand(cp))
-    # col[o] = 0.0
-    col[t] = 1.0
+    col = ntuple(i -> i == p ? cpc : Float64(i == t), Val(3))
```

Comment:
Is `srgb_compand(cpc)` always returning `Float64`? Otherwise, I think it might not be type stable.

```suggestion
    col = ntuple(i -> i == p ? cpc : typeof(cpc)(i == t), Val(3))
```